src/nrnpython/nrnpy_hoc.cpp

```diff
@@ -157,11 +159,8 @@ static int hoc_evalpointer_err() {
 }
 
 static PyObject* nrnexec(PyObject* self, PyObject* args) {
-    const char* cmd;
-    if (!PyArg_ParseTuple(args, "s", &cmd)) {
-        return NULL;
-    }
-    bool b = hoc_valid_stmt(cmd, 0);
+    auto [cmd] = nrn::cast_tuple<const std::string>(nb::tuple(args));
```

Comment:
I have noticed that some of these constructors are the equivalent of the python constructor of the same name, which is also the case here. `nb::tuple()` calls either `PySequence_Tuple` or `PyTuple_New` effectively behaving as python's `tuple()`
I have manually `steal` or `borrow` and let the object be moved 
(`nb::tuple(nb::borrow(args))` in this case). Not sure if there is anything to be gained, it's mostly for info.

Reply:
I'm not sure I follow what you mean by:

> I have manually steal or borrow and let the object be moved (nb::tuple(nb::borrow(args)) in this case)

ie: what you mean by `moved` in the above, and why would one add the extra `nb::borrow()`?

Reply:
My comment is basically, there's a lot going on in the background when you do `nb::tuple()`, whereas if do you would have done `nb::tuple(nb::borrow(args)` there's an explicit creation of `nb::object` with a borrowed reference, which being an rvalue-ref it's moved to be a `nb::tuple` in a almost-no-op.
Since I don't fully know what your direct call implicitly does, sometimes being explicit is better. But if you find out it does the very right (and minimal) thing I'd be glad to know.

Reply:
Sorry, I was on vacation and missed this.

> whereas if you would have done nb::tuple(nb::borrow(args) there's an explicit creation of nb::object with a borrowed reference

I'm all for being explicit, but I'm not sure why I want to `nb::borrow`; the refcnt doesn't need to be incremented from what I can tell, why would we need to do that?

Reply:
`nb::tuple` extends `nb::object`, there there will be decref at the end of it's life. If it's not borrowed there will be a bug. Hence my comment, maybe we would better be explicit about it.

Reply:
> nb::tuple extends nb::object, so there will be decref at the end of its life.

I don't see why that is a problem.  The `nb::tuple` calls https://github.com/wjakob/nanobind/blob/master/include/nanobind/nb_types.h#L478 which then calls https://github.com/wjakob/nanobind/blob/master/src/common.cpp#L623
This is then used to initialize the `nb::object`, so it has the correct refcount, and thus when it is dtor'd, everything is correct.

Note that both code paths go through `PySequence_Tuple`, so a new python tuple is created with its own lifetime and refcount.